lib/puppet/provider/radius_server/cisco.rb

```diff
@@ -141,9 +157,18 @@ def flush
       @radius_server = nil
       @property_hash[:ensure] = :absent
     else
-      if @property_hash.empty?
+      if @property_hash.empty? ||
+         (Facter.value('operatingsystem').eql?('ios_xr') &&
```

Comment:
Not clear to me why this special handling is needed for xr.  Can you add comments to clarify?